File: biddo_client/lib/core/repositories/report.dart
import 'package:get/get.dart';

import 'base.dart';

class ReportsRepository {
  var dio = Get.find<Api>();

  Future<bool> create(
    String entity,
    String entityId,
    String reason, [
    String description = '',
  ]) async {
    try {
      await dio.api.post('/report', data: {
        "entityName": entity,
        "entityId": entityId,
        "reason": reason,
        "description": description
      });

      return true;
    } catch (error) {
      print('error creating report: $error');
      return false;
    }
  }
}

File: core/repositories/report.ts
import { RequestMaker, RequestType } from '../services/request-maker'

class ReportsRepository {
  private basePath = '/report'

  public async create(
    entity: string,
    entityId: string,
    reason: string,
    description = ''
  ): Promise<boolean> {
    try {
      await RequestMaker.makeRequest({
        path: this.basePath,
        method: RequestType.POST,
        payload: JSON.stringify({
          entityName: entity,
          entityId: entityId,
          reason: reason,
          description: description,
        }),
        contentType: 'application/json',
      })
      return true
    } catch (error) {
      console.error('Error creating report:', error)
      return false
    }
  }
}

const ReportsRepositoryInstance = new ReportsRepository()
export { ReportsRepositoryInstance as ReportsRepository }
